models/User.js

```diff
@@ -136,4 +141,15 @@ userSchema.methods.buyBuilding = function(building, buildingCost) {
 	return this.save();
 };
 
+userSchema.methods.recruitUnits = function(unit, amount) {
+	for(const resource in unit.cost) {
+		this.resources[resource] -= unit.cost[resource] * amount;
+	}
+
```

Comment:
`this.resources.forEach(r=> {
r-=unit.cost[r] * amount;
}
`
Prøv å unngå `for in` hvor vi kan:
https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea